src/moscot/plotting/_utils.py

```diff
@@ -359,45 +360,137 @@ def _plot_temporal(
+        titles = title
     else:
-        constant_fill_keys = set(all_keys) - set(time_points)
-    tmp = np.full(len(adata), np.nan)
-    for t in adata.obs[temporal_key].unique():
-        mask = adata.obs[temporal_key] == t
-        if t in constant_fill_keys:
-            tmp[mask] = constant_fill_value
+        name = "descendants" if push else "ancestors"
+        if time_points is not None:
+            titles = [f"{categories} at time point {start if push else end}"]
+            titles.extend([f"{name} at time point {time_points[i]}" for i in range(1, len(time_points))])
         else:
-            tmp[mask] = adata[adata.obs[temporal_key] == t].obs[key_stored]
-
-    adata.obs[result_key] = tmp
-
-    sc.set_figure_params(figsize=figsize, dpi=dpi)  # TODO(@MUCDK, michalk8): necessary? want to make it uniform
-    fig = sc.pl.embedding(
-        adata=adata,
-        basis=basis,
-        color=result_key,
-        color_map=cont_cmap,
-        title=title,
-        ax=ax,
-        show=show,
-        **kwargs,
-    )
+            titles = [f"{categories} at time point {start if push else end} and {name}"]
+    for i, ax in enumerate(axs):
+        if time_points is None:
+            if scale:
+                vmin, vmax = np.nanmin(adata.obs[key_stored]), np.nanmax(adata.obs[key_stored])
+            else:
+                vmin, vmax = 0, 1
+            adata.obs[f"result_key_{i}"] = (adata.obs[key_stored] - vmin) / (vmax - vmin)
+            size = None
+        else:
+            tmp = np.full(len(adata), constant_fill_value)
+            mask = adata.obs[temporal_key] == time_points[i]
+
+            tmp[mask] = adata[adata.obs[temporal_key] == time_points[i]].obs[key_stored]
+            if scale:
+                vmin = np.nanmin(adata.obs[key_stored] * mask)
+                vmax = np.nanmax(adata.obs[key_stored] * mask)
+            else:
+                vmin = 0
+                vmax = 1
+            adata.obs[f"result_key_{i}"] = (tmp - vmin) / (vmax - vmin)
+            size = (mask * 120000 * dot_scale_factor + (1 - mask) * 120000) / adata.n_obs  # 120,000 from scanpy
+
+            _ = kwargs.pop("color_map", None)
+            _ = kwargs.pop("palette", None)
+
+            if isinstance(cont_cmap, str):
+                try:
+                    cont_cmap = mpl.colormaps[cont_cmap]
+                except KeyError:
+                    raise KeyError("`cont_cmap` is not a valid key for `matplotlib.colormaps`.")
+            if (time_points[i] == start and push) or (time_points[i] == end and not push):
+                vmin, vmax = np.nanmin(adata.obs[f"result_key_{i}"]), np.nanmax(adata.obs[f"result_key_{i}"])
+                adata.obs[f"result_key_{i}"] = adata.obs[f"result_key_{i}"].astype("str")
+                st = f"not in {time_points[i]}"
+                adata.obs[f"result_key_{i}"] = adata.obs[f"result_key_{i}"].replace("nan", st)
```

Comment:
I also dislike any plotting function writing the `adata`, unless the value is being computed and it's non-trivial to compute it (you're just scaling it), would remove this temporary keys; seel also [here](https://github.com/theislab/cellrank/blob/master/cellrank/_utils/_utils.py#L65) for a possible solution.